### PR TITLE
Cleanup references

### PR DESCRIPTION
**Pull request by davkean**
- Remove unneeded reference
- Promote "implicit" package refs
   - These packages are ones contained in RoslynTools.RepoToolset package and require our repo to double-restore before it has downloaded all the packages. This one of the causes that prevents from Visual Studio opening a non-restored repository and "just working". Promote these to real references which also gives us a good indication of what our repository depends on.
- Cleanup package sources

### CODE REVIEW

File: build/Packages.targets [hunk 1]
```diff
@@ -13,19 +13,26 @@
       https://dotnet.myget.org/F/project-system-interop/api/v3/index.json;
       https://vside.myget.org/F/vs-impl/api/v3/index.json;
       https://vside.myget.org/F/vssdk/api/v3/index.json;
+      https://api.nuget.org/v3/index.json;
```
Comment (jmarolf) [inline build/Packages.targets]: why remove this?

File: build/Packages.targets [hunk 2]
```diff
@@ -13,19 +13,26 @@
...
+    <PackageReference Update="XliffTasks"                                                             Version="$(XliffTasksVersion)" />
     <PackageReference Update="RoslynDependencies.ProjectSystem.OptimizationData"                      Version="$(RoslynDependenciesProjectSystemOptimizationDataVersion)" />
     <PackageReference Update="Microsoft.DotNet.IBCMerge"                                              Version="[$(MicrosoftDotNetIBCMergeVersion)]" />
     <PackageReference Update="Microsoft.DiaSymReader.Pdb2Pdb"                                         Version="$(MicrosoftDiaSymReaderPdb2PdbVersion)" />
+    <PackageReference Update="Microsoft.Net.Compilers"                                                Version="3.3.0-beta2-19357-02" />
     <PackageReference Update="OpenCover"                                                              Version="$(OpenCoverVersion)" />
     <PackageReference Update="Codecov"                                                                Version="$(CodecovVersion)" />
+    <PackageReference Update="MicroBuild.Core"                                                        Version="$(MicroBuildCoreVersion)" />
```
Comment (jmarolf) [inline build/Packages.targets]: Why do we need to update these?

File: build/Packages.targets (hunk 1 above)
Comment (davkean) [reply build/Packages.targets]: I didn't, this is an addition.

File: build/Packages.targets (hunk 2 above)
Comment (davkean) [reply build/Packages.targets]: See the update, I've explicitly set their version. I want to remove the double-restore - so I've moved them into our dependency graph.

File: build/Packages.targets (hunk 1 above)
Comment (jmarolf) [reply build/Packages.targets]: Sorry, just waking up

File: build/Packages.targets (hunk 2 above)
Comment (jmarolf) [reply build/Packages.targets]: All the micro-build package should be restore-able only on a signed build machine. I don't understand how setting the version of packages that will fail to restore / aren't restored fixes things

Comment (davkean) [reply build/Packages.targets]: > All the micro-build package should be restore-able only on a signed build machine
>  I don't understand how setting the version of packages that will fail to restore

Explain what you mean by this. This doesn't fail to restore.

Comment (jmarolf) [reply build/Packages.targets]: I see, `MicroBuild.Core` was uploaded to the dotnet-buildtools myget feed [here](https://dotnet.myget.org/feed/dotnet-buildtools/package/nuget/MicroBuild.Core). So restore will work. I think this won't break the signed build (which uses version 1.42) but you should check.

Comment (jmarolf) [reply build/Packages.targets]: And now that I checked on the signed build, it looks like that was broken in a recent PR...

Comment (davkean) [reply build/Packages.targets]: This one: https://github.com/dotnet/project-system/pull/5107/files would be one cause @drewnoakes.

Comment (davkean) [reply build/Packages.targets]: See: %USERPROFILE\.nuget\packages\roslyntools.repotoolset\1.0.0-beta-62705-01\tools\Tools.proj

We have these dependencies, we use it to build the Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles project.

Comment (davkean) [reply build/Packages.targets]: @jmarolf Can we modify .vsts-ci.yml to call something new, or does this require us to push that to every branch?

Comment (davkean) [reply build/Packages.targets]: Jon answered with yes, .vsts-ci.yml is per branch.

**Comment by davkean** [comment]
This passed the sign build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2871911.

**Pull request merged**